### PR TITLE
support projected name for model

### PR DESCRIPTION
**Pull request by chunyu3**
# Description
Fix https://github.com/Azure/autorest.csharp/issues/3662

- The model name will be retrieved in following orders: projectedName, friendlyName, NameWithTemplate, the origin Name

current getLibraryName of TCGC only considers projected Name. After getLibraryName adopt other factors, we will adopt getLibraryName

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: eng/pipelines/sdk-update.yml [hunk 1]
```diff
@@ -190,7 +190,6 @@ stages:
       - job: Create_PR
         dependsOn: 
           - Update_Generator_Versions
-          - Update_Code
```
Comment (lirenhe) [inline eng/pipelines/sdk-update.yml:193]: Are those pipeline changes related to the support of projected name?

File: src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts [hunk 2]
```diff
@@ -1,11 +1,20 @@
-import { Model } from "@typespec/compiler";
+import { Model, getFriendlyName, getProjectedNames } from "@typespec/compiler";
+import {
```
Comment (lirenhe) [inline src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts:2]: Do we have the code style check enabled to sort the imports?

Comment (chunyu3) [reply src/TypeSpec.Extension/Emitter.Csharp/src/lib/utils.ts:2]: Yes. there is prettier format to control this and it is a check in the pipeline.

File: eng/pipelines/sdk-update.yml (hunk 1 above)
Comment (chunyu3) [reply eng/pipelines/sdk-update.yml:193]: No. it is not related. It is to temporary solution to generate pr even the codegen fail. So we can bypass the openai customize code failure to continue the process.

**Comment by lirenhe** [comment]
As there are api change in the regen PR for OpenAI and ContentSafety,  @chunyu3, could you work with the owners to get the breakings addressed/approved?

**Comment by chunyu3** [comment]
> As there are api change in the regen PR for OpenAI and ContentSafety, @chunyu3, could you work with the owners to get the breakings addressed/approved?

Yes, I have contacted owners to review. ContentSafety owner has approved, and waiting for OpenAI response. @lirenhe

**Review by lirenhe — APPROVED** [review]

**Pull request merged**